Schutz inequality measures

sjsrey opened the pull request.

knaaptime commented on inequality/atkinson.py:
```diff
@@ -0,0 +1,112 @@
+
+    The Atkinson index is a measure of economic inequality that takes
+    into account the social aversion to inequality. The equally
+    distributed equivalent(EDE) represents the level of income that,
```
missing a space

knaaptime commented on inequality/atkinson.py:
```diff
@@ -0,0 +1,112 @@
+__all__ = ["Atkinson"]
+
+
+def _atkinson(y, epsilon):
```
this has a really similar API to segregation, where the function is private and the statistic stored internally as an attributed on the class. As we've talking about in segregation and esda, it would probably be nice to expose both function and OO api's (i.e. by tweaking the structure a bit and making `_atkinson` public).

Just dropping this note here as something for us to consider as we work on unifying standards across the meta

knaaptime commented:
(note to self: continue to always check for  'pending' status on reviews moving forward

knaaptime commented:
(also, i think you started the schutz branch from your atkinson branch, so this commit includes both additions). Can probably close #74

jGaboardi commented on inequality/schutz.py:
```diff
@@ -0,0 +1,199 @@
+    perfect equality where the Schutz distance occurs and the original
+    Schutz coefficient.
+    See :cite:`schutz1951MeasurementIncome`.
+
```
```suggestion
```

jGaboardi commented on inequality/schutz.py:
```diff
@@ -0,0 +1,199 @@
+    Schutz coefficient.
+    See :cite:`schutz1951MeasurementIncome`.
+
+
```
```suggestion
```

jGaboardi commented on inequality/schutz.py:
```diff
@@ -0,0 +1,199 @@
+    See :cite:`schutz1951MeasurementIncome`.
+
+
+
```
```suggestion
```

jGaboardi commented:
Seems like we need to increase the timeout on the testing workflow. There Mac and Windows aren't finish within 30 minutes, but they almost are...

jGaboardi commented:
Probably [bumping to 35 minutes](https://github.com/pysal/inequality/blob/36e5e8d00b4ab7d51bde1a060c726bf30d6dfa95/.github/workflows/testing.yml#L23) should do the trick.

knaaptime commented:
kinda taking awile for not a *ton* of functionality. Have we turned on parallel tests here yet?

martinfleis commented:
The timeout comes only on Mac and Win and only in this PR. Given ubuntu runs in under a minute, I smell some issue related to non-linux runtime of something in this PR. If that is the case, bumping the timeout is probably not the optimal solution but we should figure out what is causing so massive performance difference.

sjsrey commented:
My suspicion has to do with some tests I added for the plots. Trying to get the blessing of the codecov gods, I added some tests to cover, well everything. When I run them locally, separate windows actually get generated locally now (just noticed this this morning). 
I'm at a loss as to why linux can handle this, but not macos or windows on the ci. But I can revisit the tests to make them less troublesome.

sjsrey commented on inequality/tests/test_schutz.py:
```diff
@@ -0,0 +1,56 @@
+    assert pytest.approx(schutz_obj.coefficient, 0.1) == expected_coefficient
+
+
+def test_schutz_plot_runs_without_errors(example_dataframe):
```
Check if this is causing the timeout on mac and windows ci.

jGaboardi commented:
> kinda taking awile for not a _ton_ of functionality. Have we turned on parallel tests here yet?

`pytest-xdist` is enabled

martinfleis commented on inequality/schutz.py:
```diff
@@ -0,0 +1,197 @@
+import matplotlib.pyplot as plt
```
Inequality does not depend on matplotlib as of now, so this either needs to be imported in a function or we need to add matplotlib to dependencies. This is now [failing momepy CI](https://github.com/pysal/momepy/actions/runs/10658854125/job/29540556262) which is tested against dev inequality.
